File: app/src/main/java/com/ivanovsky/passnotes/data/repository/TemplateRepository.java
package com.ivanovsky.passnotes.data.repository;

import androidx.annotation.NonNull;

import com.ivanovsky.passnotes.data.entity.OperationResult;
import com.ivanovsky.passnotes.data.entity.Template;

import java.util.List;
import java.util.UUID;

import javax.annotation.Nullable;

public interface TemplateRepository {

    @Nullable
    UUID getTemplateGroupUid();

    @Nullable
    List<Template> getTemplates();

    @NonNull
    OperationResult<Boolean> addTemplates(@NonNull List<Template> templates);
}

File: app/src/main/kotlin/com/ivanovsky/passnotes/data/repository/TemplateRepository.kt
package com.ivanovsky.passnotes.data.repository

import com.ivanovsky.passnotes.data.entity.OperationResult
import com.ivanovsky.passnotes.data.entity.Template
import java.util.UUID

interface TemplateRepository {
    fun getTemplateGroupUid(): OperationResult<UUID?>
    fun getTemplates(): OperationResult<List<Template>>
    fun addTemplates(templates: List<Template>): OperationResult<Boolean>
    fun addTemplates(templates: List<Template>, doCommit: Boolean): OperationResult<Boolean>
}